pkg/chart/validate.go

```diff
@@ -0,0 +1,92 @@
+package chart
+
+import (
+	"fmt"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubescheme "k8s.io/client-go/kubernetes/scheme"
+	helmchart "k8s.io/helm/pkg/proto/hapi/chart"
+)
+
+func ValidateServices(decoded []runtime.Object) error {
+	services := make([]*corev1.Service, 0)
+	lbs := make([]*corev1.Service, 0)
+
+	for _, decodedObj := range decoded {
+		if svc, ok := decodedObj.(*corev1.Service); ok {
+			services = append(services, svc)
+			if lbValue, ok := svc.Labels[shipper.LBLabel]; ok && lbValue == shipper.LBForProduction {
+				lbs = append(lbs, svc)
+				if len(lbs) > 1 {
+					return fmt.Errorf(
+						fmt.Sprintf("Object %#v contains %q label, but %#v claims"+
+							" it is the production LB. This looks like a misconfig:"+
+							" only 1 service is allowed to be the production LB.",
+							decodedObj, shipper.LBLabel, lbs[0]))
```

Comment:
Nit: you probably don't want to dump the entire object here, GVK+name should be enough.